<!--REQUEST_ID=eb0212a7-0ab4-4a91-bac8-ae7cbf825160-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=3ceed613-ecba-440e-93a4-9f682e2f9fec)

Attempted: ldns, nodejs-10_x, tor

<details><summary>Partial log (click to expand)</summary><p>

```
Operating system: x86_64-whatever-linux2
./config: eval: line 1: /usr/bin/env: not found
building
build flags: -j2 -l2 SHELL=/nix/store/czx8vkrb9jdgjyz8qfksh10vrnqa723l-bash-4.4-p23/bin/bash MANDIR=\$\(man\)/share/man
make: *** No targets specified and no makefile found.  Stop.
builder for '/nix/store/r4da46wj7zk26vwwwfnc4gvp0n9axbx3-openssl-1.1.1.drv' failed with exit code 2
cannot build derivation '/nix/store/pqx6x1asb5qp6if2nipny8nmd0jlxdid-ldns-1.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3h5q2czc35bzyvnri5m9zybcq5j4kdpx-nodejs-10.9.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/gzr81b0xzrvmmsdq8lsgvdv7zjq39ibk-tor-0.3.4.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3h5q2czc35bzyvnri5m9zybcq5j4kdpx-nodejs-10.9.0.drv', '/nix/store/gzr81b0xzrvmmsdq8lsgvdv7zjq39ibk-tor-0.3.4.8.drv', '/nix/store/pqx6x1asb5qp6if2nipny8nmd0jlxdid-ldns-1.7.0.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, buildPackages, perl
+{ stdenv, fetchurl, buildPackages, perl, coreutils
 , withCryptodev ? false, cryptodevHeaders
 , enableSSL2 ? false
 , static ? false
@@ -31,6 +31,8 @@ let
         substituteInPlace "$a" \
           --replace /bin/rm rm
       done
+    '' + optionalString (versionAtLeast version "1.1.1") ''
+      substituteInPlace config --replace '/usr/bin/env' '${coreutils}/bin/env'
     '' + optionalString (versionAtLeast version "1.1.0" && stdenv.hostPlatform.isMusl) ''
       substituteInPlace crypto/async/arch/async_posix.h \
         --replace '!defined(__ANDROID__) && !defined(__OpenBSD__)' \

--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -125,7 +125,7 @@ in {
     sha256 = "003xh9f898i56344vpvpxxxzmikivxig4xwlm7vbi7m8n43qxaah";
   };
 
-  openssl_1_1_1 = common {
+  openssl_1_1 = common {
     version = "1.1.1";
     sha256 = "0gbab2fjgms1kx5xjvqx8bxhr98k4r8l2fa8vw7kvh491xd8fdi8";
   };

--- a/pkgs/tools/misc/powerline-rs/default.nix
+++ b/pkgs/tools/misc/powerline-rs/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, rustPlatform, fetchFromGitHub, pkgconfig, file, perl, curl, cmake, openssl_1_1, libssh2, libgit2, libzip, Security }:
+{ stdenv, lib, rustPlatform, fetchFromGitHub, pkgconfig, file, perl, curl, cmake, openssl, libssh2, libgit2, libzip, Security }:
 rustPlatform.buildRustPackage rec {
   pname = "powerline-rs";
   name = "${pname}-${version}";
@@ -15,7 +15,7 @@ rustPlatform.buildRustPackage rec {
   cargoSha256 = "184s432a6damzvl0lv6jar1iml9dq60r190aqjy44lcg938981zc";
 
   nativeBuildInputs = [ pkgconfig file perl cmake curl ];
-  buildInputs = [ openssl_1_1 libssh2 libgit2 libzip ] ++ lib.optional stdenv.isDarwin Security;
+  buildInputs = [ openssl libssh2 libgit2 libzip ] ++ lib.optional stdenv.isDarwin Security;
 
   postInstall = ''
     install -Dm 755 "${pname}.bash" "$out/etc/bash_completion.d/${pname}"

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3688,11 +3688,11 @@ with pkgs;
   nodejs-slim-8_x = callPackage ../development/web/nodejs/v8.nix { enableNpm = false; };
 
   nodejs-10_x = callPackage ../development/web/nodejs/v10.nix {
-    openssl = openssl_1_1_1;
+    openssl = openssl_1_1;
   };
   nodejs-slim-10_x = callPackage ../development/web/nodejs/v10.nix {
     enableNpm = false;
-    openssl = openssl_1_1_1;
+    openssl = openssl_1_1;
   };
 
   nodePackages_10_x = callPackage ../development/node-packages/default-v10.nix {
@@ -3728,7 +3728,7 @@ with pkgs;
   ldapvi = callPackage ../tools/misc/ldapvi { };
 
   ldns = callPackage ../development/libraries/ldns {
-    openssl = openssl_1_1_1;
+    openssl = openssl_1_1;
   };
 
   leafpad = callPackage ../applications/editors/leafpad { };
@@ -5574,7 +5574,7 @@ with pkgs;
   tokei = callPackage ../development/tools/misc/tokei { };
 
   tor = callPackage ../tools/security/tor {
-    openssl = openssl_1_1_1;
+    openssl = openssl_1_1;
     # remove this, when libevent's openssl is upgraded to 1_1_0 or newer.
     libevent = libevent.override {
       sslSupport = false;
@@ -11444,9 +11444,7 @@ with pkgs;
 
   wolfssl = callPackage ../development/libraries/wolfssl { };
 
-  openssl = openssl_1_0;
-  openssl_1_0 = openssl_1_0_2;
-  openssl_1_1 = openssl_1_1_1;
+  openssl = openssl_1_0_2;
 
   inherit (callPackages ../development/libraries/openssl {
       fetchurl = fetchurlBoot;
@@ -11456,7 +11454,7 @@ with pkgs;
       };
     })
     openssl_1_0_2
-    openssl_1_1_1;
+    openssl_1_1;
 
   openssl-chacha = callPackage ../development/libraries/openssl/chacha.nix {
     cryptodevHeaders = linuxPackages.cryptodev.override {

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15164,6 +15164,7 @@ with pkgs;
 
   powerline-rs = callPackage ../tools/misc/powerline-rs {
     inherit (darwin.apple_sdk.frameworks) Security;
+    openssl = openssl_1_1;
   };
 
   profont = callPackage ../data/fonts/profont { };

